org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/xbase/XbaseUsageDetector.java

```diff
@@ -35,7 +35,7 @@ public boolean usesXImportSection(Grammar grammar) {
 		new UsedRulesFinder(usedRules).compute(grammar);
 		for (AbstractRule it : usedRules) {
 			if ("XImportSection".equals(it.getName())
-					&& "org.eclipse.xtext.xbase.Xtype".equals(GrammarUtil.getGrammar(it).getName())) {
+					&& "http://www.eclipse.org/xtext/xbase/Xtype".equals(it.getType().getMetamodel().getEPackage().getNsURI())) {
```

Comment:
If any of these would be null, it would have been reported before.

It is possible to override rules and return a specialization of XImportSection.
You might want to try something like `EcoreUtil2.isAssignableFrom(XtypePackage.Literals.XIMPORT_SECTION, (EClass)it.getType().getClassifier())`

Reply:
HI @szarnekow ,
no, that version does not work, the it.getType().getClassifier() returns an empty list for getEAllSupperTypes, so EcoreUtil2.isAssignableFrom returns false.

Reply:
Can you confirm that the Classifier is an instanceof EClass and is resolved, e.g. not a proxy?

Reply:
Ah, this might suffer from a duality of EPackages:
XtypePackage.Literals.XIMPORT_SECTION must likely be replaced by something like `it.eResource().getResourceSet().getEObject(EcoreUtil.getURI(XtypePackage.Literals.XIMPORT_SECTION))`.

Reply:
@szarnekow ,
Thanks, that version works.
What is the duality of EPackages, by the way?

Reply:
I do not understand why the Xtype class cannot be found, do I need to use ;x-installation:=greedy in the manifest together with org.eclipse.xtext.xbase?

Reply:
I have read about the builds in https://github.com/eclipse/xtext/blob/master/Builds.md, but it is still not clear to me why the type cannot be resolved. Do you know what I need to adapt in addition to the MANIFEST.MF?

Reply:
we have a splitted repo unfortunetly. xtext-core does not know anything about downstream xbase/common.types which are in xtext-extras. we also build with gradle which does not know anything about manifest.mf
so i fear you cannot access any xbase/common.types java types by class literal

Reply:
i guess also @szarnekow has forgotten that

Reply:
True. I wrongfully assumed that the generator has a dependency on Xbase, but that’s only fake by means of the XbaseGeneratorFragment.

Reply:
OK, then I go back to the version based on the URI. That is the best we can do, right?

Reply:
If we use a hardcoded URI, I'd be in favor of hardcoding the URI for the XImportSection EClass and attempt to find that one in the resource set + do the isAssignableFrom.

Reply:
sure, no problem. I have sent that version for review.